plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePartitionedBucketFunction.java

```diff
@@ -87,7 +87,7 @@ private static long hashCodeNullSafe(MethodHandle hashCode, Block block, int pos
             throws Throwable
     {
         if (block.isNull(position)) {
-            return 0;
+            return -1;
```

Comment:
include a code comment, why it's not 0 and why it doesn;'t need to be consistent with other places where we assign 0 for null's hash

Reply:
Should we change the value of `io.trino.spi.type.TypeUtils#NULL_HASH_CODE` constant to be -1 (or some other, more "random" value)?